app/Models/Foundation/Summit/Registration/SummitTicketTypeWithPromo.php

```diff
@@ -0,0 +1,263 @@
+<?php namespace models\summit;
+/**
+ * Copyright 2023 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+class SummitTicketTypeWithPromo extends SummitTicketType
+{
+    /**
+     * @var SummitTicketType
+     */
+    protected $type;
+
+    /**
+     * @var SummitRegistrationPromoCode
+     */
+    protected $promo_code;
+
+    public function __construct(SummitTicketType $type, SummitRegistrationPromoCode $promo_code)
+    {
+        parent::__construct();
+        $this->type = $type;
+        $this->promo_code = $promo_code;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->type->getId();
+    }
+
+    /**
+     * @return int
+     */
+    public function getIdentifier()
+    {
+        return $this->type->getIdentifier();
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->type->getName();
+    }
+
+    /**
+     * @return string
+     */
+    public function getDescription()
+    {
+        return $this->type->getDescription();
+    }
+
+    /**
+     * @return string
+     */
+    public function getExternalId(): ?string
+    {
+        return $this->type->getExternalId();
+    }
+
+    /**
+     * @return float
+     */
+    public function getCost(): float
+    {
+        return $this->type->getCost();
+    }
+
+    /**
+     * @return float
+     */
+    public function getCostWithAppliedDiscount(): float
+    {
+        if ($this->promo_code instanceof SummitRegistrationDiscountCode) {
```

Comment:
please dont redo this logic
take this logic https://github.com/OpenStackweb/summit-api/blob/3f829320242f390be9fece18736ad925a79e01e3/app/Models/Foundation/Summit/Registration/PromoCodes/SummitRegistrationDiscountCode.php#L205
and extract it to use on both places

Reply:
done